server/domains/list/service.go

```diff
@@ -0,0 +1,89 @@
+package list
+
+import (
+	"database/sql"
+
+	"github.com/h3poteto/fascia/server/domains/list_option"
+	"github.com/pkg/errors"
+)
+
+// Find returns a list entity
+func Find(targetProjectID, targetListID int64, infrastructure Repository) (*List, error) {
+	id, projectID, userID, title, color, optionID, isHidden, err := infrastructure.Find(targetProjectID, targetListID)
+	if err != nil {
+		return err
+	}
+	return New(id, projectID, userID, title, color, optionID, isHidden, infrastructure), nil
+}
+
+// Lists returns all lists related a project.
+func Lists(targetProjectID int64, infrastructure Repository) ([]*List, error) {
+	var result []*List
+
+	lists, err := infrastructure.Lists(targetProjectID)
+	if err != nil {
+		return result, err
+	}
+	for _, list := range lists {
+		id, ok := list["id"].(int64)
+		if !ok {
+			return nil, errors.New("Can not convert interface")
+		}
+		projectID, ok := list["projectID"].(int64)
+		if !ok {
+			return nil, errors.New("Can not convert interface")
+		}
+		userID, ok := list["userID"].(int64)
+		if !ok {
+			return nil, errors.New("Can not convert interface")
+		}
+		title, ok := list["title"].(sql.NullString)
+		if !ok {
+			return nil, errors.New("Can not convert interface")
+		}
+		color, ok := list["color"].(sql.NullString)
+		if !ok {
+			return nil, errors.New("Can not convert interface")
+		}
+		optionID, ok := list["optionID"].(sql.NullInt64)
+		if !ok {
+			return nil, errors.New("Can not convert interface")
+		}
+		isHidden, ok := list["isHidden"].(bool)
+		if !ok {
+			return nil, errors.New("Can not convert interface")
+		}
+		l := New(id, projectID, userID, title, color, optionID, isHidden, infrastructure)
+		resutl = append(result, l)
+	}
+	return resutl, nil
+}
+
+// NoneList retruns a none list related a project.
+func NoneList(targetProjectID int64, infrastructure Repositroy) (*List, error) {
+	id, projectID, userID, title, color, optionID, isHidden, err := infrastructure.NoneList(targetProjectID)
+	if err != nil {
+		return err
+	}
+	return New(id, projectID, userID, title, color, optionID, isHidden, infrastructure), nil
+}
+
+func (l *List) ListOption(infrastructure list_option.Repository) (*list_option.ListOption, error) {
+	if !l.ListOptionID.Valid {
+		return nil, errors.New("list has no list option")
+	}
+	option, err := list_option.FindByID(l.ListOptionID.Int64, infrastructure)
+	if err != nil {
+		return nil, err
+	}
+	return option, nil
+}
+
+// // HasCloseAction check a list has close list option
```

Comment:
__[Golint]__ comment on exported method List.HasCloseAction should be of the form "HasCloseAction ..."

<sub>posted by Sider</sub>